coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -657,7 +670,36 @@ class MultiPartitionPlanner(partitionLocationProvider: PartitionLocationProvider
             (Some(next), ep)
         }
       }
-      execPlans
+      // Edge case with namespace across multiple partitions includes a case where the a part of time range
+      // of the original params fall in the new partition but that duration is less than the lookback. For example,
+      // Consider the timesplit happening at 1:00, the lookback of the query is 20 mins and the original query params
+      // have end date of 1:10. Then the period of 1 - 1:20 will not have any results, this is due to that fact the
+      // period 1 - 1:10 can not be served from one partition alone and needs to be computed on query service. Here
+      // we will handle this case and add the missing execPlan if needed
+
+      if (partitions.length > 1 && queryConfig.routingConfig.supportRemoteRawExport) {
+        // Here we check if the assignment ranges cover the entire query duration
+        val (_, lastTimeRange) = assignmentRanges.last
+        if (lastTimeRange.endMs < timeRange.endMs) {
+          // this means we need to add the missing time range to the end to execute the bit on Query service
+          val (gapStartTimeMs, gapEndTimeMs) = stepMsOpt match {
+            case Some(step)   =>   (snapToStep(lastTimeRange.endMs + 1, step, timeRange.startMs),
```

Comment:
Why don't these timestamps need the same `* 1000 / 1000 + 1` treatment as before?

Reply:
Because we are doing snapToStep now which will take care to align the start exactly at the time where it needs to start.